ospd_openvas/openvas_db.py

```diff
@@ -60,219 +44,234 @@
-    global DB_INDEX
-    if get_db_connection() is 2:
-        return 2
-    try:
-        ctx = redis.Redis(unix_socket_path=DB_ADDRESS,
-                          db=dbnum,
-                          socket_timeout=SOCKET_TIMEOUT, charset="latin-1",
-                          decode_responses=True)
-    except ConnectionError as e:
-        return {"error": str(e)}
-    DB_INDEX = dbnum
-    return ctx
-
-def db_find(patt):
-    """ Search a pattern inside all kbs. When find it return it.
-    """
-    for i in range(0, MAX_DBINDEX):
-        ctx = kb_connect (i)
-        if ctx.keys(patt):
-            return ctx
-
-def kb_new():
-    """ Return a new kb context to an empty kb.
-    """
-    ctx = db_find(DBINDEX_NAME)
-    for index in range(1, MAX_DBINDEX):
-            if try_database_index(ctx, index) == 1:
-                ctx = kb_connect(index)
+
+class OpenvasDB(object):
+    """ Class to connect to redis, to perform queries, and to move
+    from a KB to another."""
+
+    def __init__(self):
+        # Path to the Redis socket.
+        self.db_address = ""
+
+        # Name of the namespace usage bitmap in redis.
+        self.dbindex_name = "GVM.__GlobalDBIndex"
+        self.max_dbindex = 0
+        self.db_index = 0
+        self.rediscontext = None
+
+    def get_db_connection(self):
+        """ Retrieve the db address from openvassd config.
+        """
+        result = subprocess.check_output(
+            ['openvassd', '-s'], stderr=subprocess.STDOUT)
+
+        path = None
+        if result:
+            result = result.decode('ascii')
+            for conf in result.split('\n'):
+                if conf.find('db_address') == 0:
+                    path = conf.split('=')
+                    break
+
+        if not path:
+            raise OSPDOpenvasError('Not possible to find the path to '
+                               'the redis socket.')
+
+        self.db_address = str.strip(path[1])
+
+
+    def max_db_index(self):
+        """Set the number of databases have been configured into kbr struct.
+        """
+        ctx = self.kb_connect()
+        resp = ctx.config_get('databases')
+
+        if len(resp) == 1:
+            self.max_dbindex = int(resp.get('databases'))
+        else:
+            raise OSPDOpenvasError('Redis Error: Not possible '
+                         'to get max_dbindex.')
+
+    def set_redisctx(self, ctx):
+        """ Set the current rediscontext.
+        """
+        self.rediscontext = ctx
+
+    def db_init(self):
+        """ Set db_address and max_db_index. """
+        self.get_db_connection()
+        self.max_db_index()
+
+    def try_database_index(self, ctx, i):
+        """ Check if it is already in use. If not set it as in use and return.
+        """
+        try:
+            resp = ctx.hsetnx(self.dbindex_name, i, 1)
+        except:
+            return 2
+
+        if isinstance(resp, int) is False:
+            return 2
+
+        if resp == 1:
+            return 1
```

Comment:
What's the meaning of the return values 1 and 2? I suppose it should not return None so please add a default return value.